File: riveredge-backend/src/core/config/industry_app_catalog.py
"""
行业应用清单（免费版 / 付费版唯一真源）。

- 免费版：主仓 riveredge 发布，无需 License Key
- 付费版：私仓 kuaigeyun-pro 维护，启用需 License Key（与专业版应用同一套授权流程）

与前端 industryAppCatalog.ts 保持一致。
"""

from __future__ import annotations

from typing import FrozenSet

FREE_INDUSTRY_APP_CODES: FrozenSet[str] = frozenset(
    {
        "spoke-wheel",
    }
)

# 行业免费版原作者 GitHub（与 manifest author / author_github 一致）
FREE_INDUSTRY_AUTHOR_GITHUB: dict[str, str] = {
    "spoke-wheel": "xyt123lyq",
}

PRO_INDUSTRY_APP_CODES: FrozenSet[str] = frozenset(
    {
        "kuaimachinery",
        "kuaimolding",
        "kuaielectronics",
        "kuaiautoparts",
        "kuaimedical",
        "kuaifood",
        "kuaipackaging",
        "kuaihardware",
        "kuaidiecasting",
        "kuaiwiring",
        "kuaimotor",
        "kuaibattery",
        "kuainewequipment",
        "kuaisheetmetal",
        "kuaimold",
        "kuaisemiconductor",
    }
)

ALL_INDUSTRY_APP_CODES: FrozenSet[str] = FREE_INDUSTRY_APP_CODES | PRO_INDUSTRY_APP_CODES


def is_free_industry_app_code(app_code: str | None) -> bool:
    return str(app_code or "") in FREE_INDUSTRY_APP_CODES


def is_pro_industry_app_code(app_code: str | None) -> bool:
    return str(app_code or "") in PRO_INDUSTRY_APP_CODES


def is_industry_app_code(app_code: str | None) -> bool:
    return str(app_code or "") in ALL_INDUSTRY_APP_CODES


def requires_pro_license_for_app(app_code: str | None, *, is_pro: bool = False) -> bool:
    """专业版应用或付费行业应用启用前需 License Key。"""
    if is_pro:
        return True
    return is_pro_industry_app_code(app_code)

File: riveredge-frontend/src/pages/system/applications/industryAppCatalog.ts
/**
 * 行业应用清单（免费版 / 付费版）。
 *
 * - 免费版：主仓发布，应用中心「行业」分类展示，无需 License Key
 * - 付费版：私仓 kuaigeyun-pro compose 后入库展示，启用需 License Key（无占位卡片）
 *
 * 与 core/config/industry_app_catalog.py 保持一致。
 */

export const FREE_INDUSTRY_APP_CODES = ['spoke-wheel'] as const;

export type FreeIndustryAppCode = (typeof FREE_INDUSTRY_APP_CODES)[number];

/** 付费行业应用 code（私仓实装后入库；用于分类与 License 判定） */
export const PRO_INDUSTRY_APP_CODES = [
  'kuaimachinery',
  'kuaimolding',
  'kuaielectronics',
  'kuaiautoparts',
  'kuaimedical',
  'kuaifood',
  'kuaipackaging',
  'kuaihardware',
  'kuaidiecasting',
  'kuaiwiring',
  'kuaimotor',
  'kuaibattery',
  'kuainewequipment',
  'kuaisheetmetal',
  'kuaimold',
  'kuaisemiconductor',
] as const;

export type ProIndustryAppCode = (typeof PRO_INDUSTRY_APP_CODES)[number];

export const ALL_INDUSTRY_APP_CODES = [
  ...FREE_INDUSTRY_APP_CODES,
  ...PRO_INDUSTRY_APP_CODES,
] as const;

export const FREE_INDUSTRY_SORT_ORDER: Record<FreeIndustryAppCode, number> = {
  'spoke-wheel': 300,
};

/** 侧栏行业包容器（不在应用中心展示） */
export const INDUSTRY_PACK_APP_CODE = 'industry-pack';

export function isIndustryPackShellCode(code: string | undefined | null): boolean {
  return String(code || '') === INDUSTRY_PACK_APP_CODE;
}

export function shouldHideFromApplicationCenter(code: string | undefined | null): boolean {
  return isIndustryPackShellCode(code);
}

/** 行业免费版原作者 GitHub ID（与 manifest author / author_github 一致） */
export const FREE_INDUSTRY_AUTHOR_GITHUB: Record<FreeIndustryAppCode, string> = {
  'spoke-wheel': 'xyt123lyq',
};

export function isFreeIndustryAppCode(code: string | undefined | null): boolean {
  return (FREE_INDUSTRY_APP_CODES as readonly string[]).includes(String(code || ''));
}

export function isProIndustryAppCode(code: string | undefined | null): boolean {
  return (PRO_INDUSTRY_APP_CODES as readonly string[]).includes(String(code || ''));
}

export function isIndustryAppCode(code: string | undefined | null): boolean {
  return isFreeIndustryAppCode(code) || isProIndustryAppCode(code);
}

/** 启用前是否需要 License Key（专业版或付费行业） */
export function requiresProLicense(app: {
  code?: string;
  is_pro?: boolean;
}): boolean {
  if (app?.is_pro) return true;
  return isProIndustryAppCode(app?.code);
}

/** 行业免费包原作者 GitHub 用户名（应用中心编码行展示） */
export function resolveFreeIndustryAuthorUsername(
  code: string | undefined | null,
): string | undefined {
  if (!isFreeIndustryAppCode(code)) return undefined;
  return FREE_INDUSTRY_AUTHOR_GITHUB[code as FreeIndustryAppCode];
}
